packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1594,30 +1593,32 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         const envelope = message.contents as IEnvelope;
         const transformed = { ...message, contents: envelope.contents };
         const context = this.datastoreContexts.get(envelope.address);
+        assert(!!context, "There should be a store context for the op");
         context.process(transformed, local, localMessageMetadata);
     }
 
     private bindFluidDataStore(fluidDataStoreRuntime: IFluidDataStoreChannel): void {
         this.verifyNotClosed();
-        assert(this.datastoreContexts.notBoundContexts.has(fluidDataStoreRuntime.id),
-            "Store to be bound should be in not bounded set");
-        this.datastoreContexts.notBoundContexts.delete(fluidDataStoreRuntime.id);
-        const context = this.datastoreContexts.get(fluidDataStoreRuntime.id) as LocalFluidDataStoreContext;
+
+        const id = fluidDataStoreRuntime.id;
+        this.datastoreContexts.notifyOnBeforeBind(id);
```

Comment:
>notifyOnBeforeBind [](start = 31, length = 18)

this can just return the LocalFluidDataStoreContext, rather than the get and unsafe cast below